dataclients/kubernetes/kube.go

```diff
@@ -79,37 +92,130 @@ type Options struct {
 
 // Client is a Skipper DataClient implementation used to create routes based on Kubernetes Ingress settings.
 type Client struct {
+	httpClient         *http.Client
 	apiURL             string
 	provideHealthcheck bool
+	token              string
 	current            map[string]*eskip.Route
 }
 
 var nonWord = regexp.MustCompile("\\W")
 
 var (
-	errServiceNotFound     = errors.New("service not found")
-	errServicePortNotFound = errors.New("service port not found")
+	errServiceNotFound      = errors.New("service not found")
+	errServicePortNotFound  = errors.New("service port not found")
+	errAPIServerURLNotFound = errors.New("kubernetes API server URL could not be constructed from env vars")
+	errInvalidCertificate   = errors.New("discovered certificate is invalid")
```

Comment:
The failure message is missleading. Why it is "discovered"? I would suggest something like "invalid CA".